internal/pkg/ecs/ecs.go

```diff
@@ -196,21 +251,26 @@ func (c Client) listActiveCopilotTasks(opts listActiveCopilotTasksOpts) ([]*ecs.
 		}
 		tasks = resp
 	}
-	return filterCopilotTask(tasks, opts.TaskID), nil
+	return filterCopilotTask(tasks, opts.TaskID, opts.OneOffOnly), nil
 }
 
-func filterCopilotTask(tasks []*ecs.Task, taskID string) []*ecs.Task {
+func filterCopilotTask(tasks []*ecs.Task, taskID string, oneOffOnly bool) []*ecs.Task {
```

Comment:
nit: The recommendation is to create multiple functions instead of using flag arguments (https://martinfowler.com/bliki/FlagArgument.html) so for example we can do:

```go
func filterTasksByID(tasks []*ecs.Task, taskID string) []*ecs.Task { ... }

func filterCopilotTasks(tasks []*ecs.Task, taskID string) []*ecs.Task {
   var filteredTasks []*ecs.Task
   for _, t  := range filterTasksByID(tasks, taskID) {
      // loop through the tags.
   }
   ...
}
```